File: _includes/code/embedded.ts
// Blog: 2023-06-06, embedded-local-weaviate
// run with: node --loader=ts-node/esm embedded.ts
// START 10lines  // START TestExample
import weaviate, { EmbeddedOptions } from 'weaviate-ts-embedded';

const client = weaviate.client(new EmbeddedOptions(
  // END 10lines  // START TestExample
  {
    env: {
      'OPENAI_APIKEY': 'YOUR-OPENAI-API-KEY',
    },
  }
  // END TestExample  // START 10lines
));

await client.embedded.start();

const createdObject = await client.data
  .creator()
  .withClassName('MyClass')
  .withProperties({
    hello: 'World!',
  })
  .do();

console.log(JSON.stringify(createdObject, null, 2));

// END 10lines

// Test
import assert from 'assert';
assert.equal(createdObject?.properties?.['hello'], 'World!');

// START TestExample

await client.schema.classCreator().withClass({
  class: 'Wine',
  vectorizer: 'text2vec-openai',
}).do();

await client.data.creator()
  .withClassName('Wine')
  .withProperties({
    name: 'Chardonnay',
    review: 'Goes well with fish!',
  })
  .do();

const response = await client.graphql
  .get()
  .withClassName('Wine')
  .withNearText({
    concepts: ['great for seafood'],
  })
  .withFields('name review')
  .do();

test('Test nearText', function () {
  assert.equal(
    response.data.Get['Wine'][0]['review'],
    'Goes well with fish!'
  );
});

// END TestExample  // START 10lines  // START TestExample
await client.embedded.stop();
// END 10lines  // END TestExample

File: _includes/code/embedded.py
# Blog: 2023-06-06, embedded-local-weaviate
# START 10lines  # START TestExample
import weaviate
import json

client = weaviate.Client(
    embedded_options=weaviate.embedded.EmbeddedOptions(),
    # END 10lines  # START TestExample
    additional_headers={
        'X-OpenAI-Api-Key': 'YOUR-OPENAI-API-KEY'  # Replace w/ your OPENAI API key
    }
    # START 10lines  # START TestExample
)
# END TestExample

uuid = client.data_object.create({
    'hello': 'World!'
}, 'MyClass')

obj = client.data_object.get_by_id(uuid, class_name='MyClass')

print(json.dumps(obj, indent=2))
# END 10lines
client.schema.delete_all()
# Test
assert obj['properties']['hello'] == 'World!'

# START TestExample

client.schema.create_class({
    'class': 'Wine',
    'vectorizer': 'text2vec-openai',
})

client.data_object.create({
    'name': 'Chardonnay',
    'review': 'Goes well with fish!',
}, 'Wine')

response = (
    client.query
    .get('Wine', ['name', 'review'])
    .with_near_text({
        'concepts': ['great for seafood']
    })
    .do()
)

assert response['data']['Get']['Wine'][0]['review'] == 'Goes well with fish!'
# END TestExample
